File: app/src/main/java/com/tes/my/UserDao.java
package com.tes.my;

import androidx.room.*;
import java.util.List;

/**
 * UserDao — Data Access Object untuk User.
 * TODO: Add queries as needed
 */
@Dao
public interface UserDao {

    @Insert
    void insert(User user);

    @Query("SELECT * FROM users ORDER BY id ASC")
    List<User> getAll();

    @Query("SELECT * FROM users WHERE id = :id LIMIT 1")
    User findById(int id);

    @Update
    void update(User user);

    @Delete
    void delete(User user);

    @Query("DELETE FROM users")
    void deleteAll();
}

File: app/src/main/java/com/tes/kk/UserDao.kt
package com.tes.kk

import androidx.room.*

/**
 * UserDao — Data Access Object untuk User.
 * TODO: Add queries as needed
 */
@Dao
interface UserDao {

    @Insert
    suspend fun insert(user: User)

    @Query("SELECT * FROM users ORDER BY id ASC")
    fun getAll(): List<User>

    @Query("SELECT * FROM users WHERE id = :id LIMIT 1")
    fun findById(id: Int): User?

    @Update
    suspend fun update(user: User)

    @Delete
    suspend fun delete(user: User)

    @Query("DELETE FROM users")
    suspend fun deleteAll()
}
